main.c

```diff
@@ -134,7 +138,8 @@ main(int argc, char *argv[])
 
 	/* Prompt for a password. */
 	if (readpass(&passwd, "Please enter passphrase",
-	    dec ? NULL : "Please confirm passphrase", 1))
+	    dec ? NULL : !devtty ? NULL :
```

Comment:
Graham Percival writes ("Re: [scrypt] Provide -P option to read password from stdin (#34)"):
> > @@ -134,7 +138,8 @@ main(int argc, char *argv[])
> >  
> >  	/* Prompt for a password. */
> >  	if (readpass(&passwd, "Please enter passphrase",
> > -	    dec ? NULL : "Please confirm passphrase", 1))
> > +	    dec ? NULL : !devtty ? NULL :
> 
> It looks like we want `confirmprompt` (the 3rd argument of `readwd()`) to be `NULL`.  From `readpass.h`:

Right, I think that's what I was thinking.

> >  If ${confirmprompt} is non-NULL, read a second password (prompting if a terminal is being used) and repeat until the user enters the same password twice.
> 
> That's born out in the code:
> ```
>         /* Confirm the password if necessary. */
>         if (confirmprompt != NULL) {
>                 if (usingtty)
>                         fprintf(stderr, "%s: ", confirmprompt);
>                 if (fgets(confpassbuf, MAXPASSLEN, readfrom) == NULL) {
>                         warnp("Cannot read password");
>                         goto err3;
>                 }
>                 if (strcmp(passbuf, confpassbuf)) {
>                         fprintf(stderr,
>                             "Passwords mismatch, please try again\n");
>                         goto retry;
>                 }
>         }
> ```
> 
> I don't see the point of sending a password over `stdin` twice, so I think we should keep that `!devtty ? NULL`.

Right.  Certainly scrypt(1)'s -P option shouldn't read the pw twice.

Arguably that readpass does so in the non-tty case is undesirable but
I guess you don't want to change that.

Thanks,
Ian.